src/akushon/action/node/action_manager.cpp

```diff
@@ -160,6 +192,48 @@ void ActionManager::start(std::string action_name, const Pose & initial_pose)
   is_running = true;
 }
 
+void ActionManager::start(std::string action_name, std::string target_action_name, const Pose & initial_pose, float ball_x, float right_map_x_min_, float right_map_x_max_, float left_map_x_min_, float left_map_x_max_, bool right)
+{
+  std::vector<Action> target_actions;
+
+  while (true) {
+    auto action = actions.at(action_name);
+    const auto & target_action = actions.at(target_action_name);
+    int pose_count = action.get_pose_count();
+    for (int pose_index = 0; pose_index < pose_count; pose_index++) {
+      if (right)
+      {
+        RCLCPP_INFO(rclcpp::get_logger("DEBUG ACTION MANAGER"), "Start Mapping!");
+        action.map_action(action, target_action, pose_index, ball_x, right_map_x_min_, right_map_x_max_);
+      } else {
+        RCLCPP_INFO(rclcpp::get_logger("DEBUG ACTION MANAGER"), "Start Mapping!");
+        action.map_action(action, target_action, pose_index, ball_x, left_map_x_min_, left_map_x_max_);
+      }
+    }
```

Comment:
If the `if` condition doesn't change inside the for-loop, it's better to do the for-loop inside the `if` condition instead.